src/main/scala/scalafix/sbt/ScalafixPlugin.scala

```diff
@@ -223,11 +229,11 @@ object ScalafixPlugin extends AutoPlugin {
   override lazy val projectConfigurations: Seq[Configuration] =
     Seq(ScalafixConfig)
 
-  override lazy val projectSettings: Seq[Def.Setting[_]] = Def.settings(
+  override lazy val projectSettings: Seq[Def.Setting[?]] = Def.settings(
     Seq(Compile, Test).flatMap(c => inConfig(c)(scalafixConfigSettings(c))),
     inConfig(ScalafixConfig)(
       Def.settings(
-        Defaults.configSettings,
+        Defaults.configSettings ++ SemanticdbPlugin.configurationSettings,
```

Comment:
sbt 2.0.0-M4 rightfully detects `Scalafix / semanticdbTargetRoot` as unused, because the key is defined in Defaults but the plugin config settings were not applied. It doesn't hurt to trigger semanticdb on scalafix rules (for metals for example)...